File: app/src/main/java/io/github/hrilab/pepper_realtime/robot/RobotLifecycleBridge.java
package io.github.hrilab.pepper_realtime.robot;

import android.app.Activity;

/**
 * Bridge interface for robot lifecycle management.
 * Flavor-specific implementations handle QiSDK registration (Pepper) or simulation (Standalone).
 */
public interface RobotLifecycleBridge {
    
    /**
     * Register for robot lifecycle events
     * @param activity The activity to register
     * @param listener Callback for lifecycle events
     */
    void register(Activity activity, RobotLifecycleListener listener);
    
    /**
     * Unregister from robot lifecycle events
     * @param activity The activity to unregister
     */
    void unregister(Activity activity);
    
    /**
     * Get the robot controller
     */
    RobotController getRobotController();
    
    /**
     * Callback interface for robot lifecycle events
     */
    interface RobotLifecycleListener {
        /**
         * Called when robot is ready (focus gained or simulation initialized)
         */
        void onRobotReady(Object robotContext);
        
        /**
         * Called when robot focus is lost (Pepper) or simulation paused (Standalone)
         */
        void onRobotFocusLost();
    }
}

File: app/src/main/java/ch/fhnw/pepper_realtime/robot/RobotLifecycleBridge.kt
package ch.fhnw.pepper_realtime.robot

import android.app.Activity

/**
 * Bridge interface for robot lifecycle management.
 * Flavor-specific implementations handle QiSDK registration (Pepper) or simulation (Standalone).
 */
interface RobotLifecycleBridge {

    /**
     * Register for robot lifecycle events
     * @param activity The activity to register
     * @param listener Callback for lifecycle events
     */
    fun register(activity: Activity, listener: RobotLifecycleListener)

    /**
     * Unregister from robot lifecycle events
     * @param activity The activity to unregister
     */
    fun unregister(activity: Activity)

    /**
     * Get the robot controller
     */
    fun getRobotController(): RobotController

    /**
     * Callback interface for robot lifecycle events
     */
    interface RobotLifecycleListener {
        /**
         * Called when robot is ready (focus gained or simulation initialized)
         */
        fun onRobotReady(robotContext: Any?)

        /**
         * Called when robot focus is lost (Pepper) or simulation paused (Standalone)
         */
        fun onRobotFocusLost()
    }
}

